bin/reth/src/node/mod.rs

```diff
@@ -263,19 +269,32 @@ impl<Ext: RethCliExt> NodeCommand<Ext> {
         // setup the blockchain provider
         let factory = ProviderFactory::new(Arc::clone(&db), Arc::clone(&self.chain));
         let blockchain_db = BlockchainProvider::new(factory, blockchain_tree.clone())?;
-
         let blob_store = InMemoryBlobStore::default();
-        let transaction_pool = reth_transaction_pool::Pool::eth_pool(
+
+        let validator = if let Some(ref trusted_setup_file) = self.trusted_setup_file {
+            let kzg_settings = self.load_kzg_settings(trusted_setup_file)?;
+            info!(target: "reth::cli", "KZG trusted setup file loaded");
+
+            TransactionValidationTaskExecutor::eth_with_kzg_settings_and_tasks(
+                blockchain_db.clone(),
+                Arc::clone(&self.chain),
+                blob_store.clone(),
+                ctx.task_executor.clone(),
+                1,
+                kzg_settings,
```

Comment:
even if this a bit verbose now, I still think this is reasonable, but could optionally convert this to a builder-style init